### PR TITLE
NuttX work towards Werror

### PR DESCRIPTION
**Pull request by dagar**
I got tired of all the warnings and unnecessary output that fly by ignored in a typical build. Fairly often I have to help someone locate the cause of a build failure, so I'm trying to make it as obvious as possible.

This PR syncs up the warnings of all nuttx-configs (nuttx-configs/PX4_Warnings.mk) and ignores the current warnings. If we merge https://github.com/PX4/NuttX/pull/70 Werror can be enabled and they can be slowly re-enabled if desired. **There shouldn't be any actual changes to the resulting Firmware.**

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Can you touch a typical file (e.g. the FW controller) and share the incremental output of this vs. the current output?

**Comment by dagar** [comment]
Incremental is similar. This is mostly about the amount of distracting output on travis-ci with 6 NuttX builds and enabling Werror.

I'm using ninja and have GIT_SUBMODULES_ARE_EVIL=1

Fresh full build, incremental build with no work, then incremental with fw_att_control_main.cpp change.
This: https://gist.github.com/dagar/7e3ed448e3db61ad5766d79e1d453631
Master: https://gist.github.com/dagar/f23bb70f973b4faa767241b8db9a407d

**Comment by dagar** [comment]
The build for mindpx-v2 was broken, but no one noticed. One example of why little things like this can be important.

**Comment by dagar** [comment]
You can now use `make check` to build all the same configurations and run the same tests as travis-ci.

File: cmake/toolchains/Toolchain-native.cmake [hunk 1]
```diff
@@ -3,7 +3,7 @@ foreach(tool nm ld)
 	string(TOUPPER ${tool} TOOL)
 	find_program(${TOOL} ${tool})
 	if(NOT ${TOOL})
-		message(FATAL_ERROR "could not find ${tool}")
+		#message(FATAL_ERROR "could not find ${tool}")
```
Comment (nicolaerosia) [inline cmake/toolchains/Toolchain-native.cmake]: This one was fatal...

Comment (LorenzMeier) [inline cmake/toolchains/Toolchain-native.cmake]: Indeed, I'll fix it, thanks!

Comment (LorenzMeier) [inline cmake/toolchains/Toolchain-native.cmake]: Fixed in 07d926b.

**Comment by dagar** [comment]
Thanks @nicolaerosia

**Pull request merged**